plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreExecutionContext.java

```diff
@@ -234,34 +235,32 @@ public List<String> getSearchPath() {
         return searchPath;
     }
 
-    List<String> getDefaultSearchPath() {
-        return defaultSearchPath;
-    }
-
     private void setSearchPath(DBRProgressMonitor monitor, PostgreSchema schema) throws DBCException {
         // Construct search path from current search path but put default schema first
         setSearchPath(monitor, schema.getName());
     }
 
     private void setSearchPath(DBRProgressMonitor monitor, String defSchemaName) throws DBCException {
-        List<String> newSearchPath = new ArrayList<>(getDefaultSearchPath());
-        int schemaIndex = newSearchPath.indexOf(defSchemaName);
-        /*if (schemaIndex == 0 || (schemaIndex == 1 && isUserFirstInPath(newSearchPath))) {
-            // Already default schema
+        LinkedList<String> newSearchPath = new LinkedList<>(getSearchPath());
+
+        if (!newSearchPath.isEmpty() && defSchemaName.equals(newSearchPath.getFirst())) {
             return;
-        } else*/
-        {
-            if (schemaIndex > 0) {
-                // Remove from previous position
-                newSearchPath.remove(schemaIndex);
-            }
-            // Add it first
-            newSearchPath.addFirst(defSchemaName);
         }
 
+        newSearchPath.remove(defSchemaName);
+        newSearchPath.addFirst(defSchemaName);
+
+        if (activeUser != null && !newSearchPath.contains(activeUser)) {
+            newSearchPath.add(activeUser);
+        }
+
+        setUserInTheEndOfThePath(newSearchPath);
```

Comment:
Why? Previously we added user only during connection init, why it is needed for every path change now?